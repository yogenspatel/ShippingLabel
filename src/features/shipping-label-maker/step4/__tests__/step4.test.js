/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import ShippingLabelStep4 from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Step 4 Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShippingLabelStep4 />, div);
  });
  it("has called snapshot with ShippingLabel component", () => {
    const tree = shallow(<ShippingLabelStep4 />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("has called componentDidMount method", () => {
    const spy = jest.spyOn(ShippingLabelStep4.prototype, 'componentDidMount');
    const props = {
        setShippingData: {},
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep4 {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
    expect(props.getShippingData).toBeCalled();
    spy.mockClear();
  });
  it("has retrieved data on componentDidMount method", () => {
    const shippingData = {
        shippingOption: {
            shippingOption: 2
        }
    }
    const spy = jest.spyOn(ShippingLabelStep4.prototype, 'componentDidMount');
    const props = {
        setShippingData: shippingData,
        getShippingData: jest.fn()
    }
    const wrapper = shallow(<ShippingLabelStep4 {...props} />);
    wrapper.instance().componentDidMount();
    expect(wrapper.state('shippingOption')).toEqual(shippingData.shippingOption.shippingOption);
    spy.mockClear();
  });
});
