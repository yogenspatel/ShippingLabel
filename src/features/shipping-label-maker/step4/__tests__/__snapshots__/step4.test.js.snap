// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 4 Component renders correctly has called snapshot with ShippingLabel component 1`] = `
<div
  className="card"
>
  <div
    className="h4 card-header"
  >
    Select the Shipping Option
  </div>
  <div
    className="card-body"
  >
    <div
      className="form-check"
    >
      <input
        checked={true}
        name="shippingOption_1"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="form-check-label ml-2"
      >
        Ground
      </label>
    </div>
    <div
      className="form-check"
    >
      <input
        checked={false}
        name="shippingOption_2"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="form-check-label ml-2"
      >
        Priority
      </label>
    </div>
  </div>
</div>
`;
